pkg/asset/tls/ironictls.go

```diff
@@ -0,0 +1,72 @@
+package tls
+
+import (
+	"context"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"errors"
+	"net"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	baremetaltypes "github.com/openshift/installer/pkg/types/baremetal"
+)
+
+// IronicTLSCert is the asset that generates the key/cert pair that is used for
+// enabling TLS for virtual media in ironic.
+type IronicTLSCert struct {
+	SelfSignedCertKey
+}
+
+var _ asset.WritableAsset = (*IronicTLSCert)(nil)
+
+// Dependencies returns the dependency of the the cert/key pair, which includes
+// the parent CA, and install config if it depends on the install config for
+// DNS names, etc.
+func (a *IronicTLSCert) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.InstallConfig{},
+	}
+}
+
+// Generate generates the cert/key pair based on its dependencies.
+func (a *IronicTLSCert) Generate(ctx context.Context, dependencies asset.Parents) error {
+	installConfig := &installconfig.InstallConfig{}
+	dependencies.Get(installConfig)
+	hostname := internalAPIAddress(installConfig.Config)
+
+	if installConfig.Config.Platform.Name() != baremetaltypes.Name {
+		return errors.New("IronicTLSCert can only be used with the Baremetal platform")
```

Comment:
Done!

Reply:
Yeah, there's no way to make optional dependencies, because we don't have access to the dependencies in the Dependencies() functions.
What we _can_ do though is not add any files to the asset in Generate() if the platform is not baremetal.